File: calico-us-berkeley/2025-fall/final/earthquake/templates/earthquake_template.cpp
#include <iostream>
#include <vector>
#include <array>
#include <string>

struct Person {
    double X, Y, S;
    std::string R;
};

std::array<int, 4> solve(const std::vector<Person>& persons) {
    // YOUR CODE HERE
    return {0, 0, 1, 1};
}

int main() {
    int T;
    std::cin >> T;

    for (int t = 0; t < T; ++t) {
        int N;
        std::cin >> N;

        std::vector<Person> persons(N);
        for (int i = 0; i < N; ++i) {
            std::cin >> persons[i].X
                     >> persons[i].Y
                     >> persons[i].S
                     >> persons[i].R;
        }

        const auto [floor_xE, floor_yE, ceil_xE, ceil_yE] = solve(persons);
        std::cout << floor_xE << " "
                  << floor_yE << " "
                  << ceil_xE << " "
                  << ceil_yE << "\n";
    }

    return 0;
}

File: calico-us-berkeley/2025-fall/final/earthquake/templates/earthquake_template.py
def solve(N: int, persons: list[tuple[float, float, float, str]]) -> tuple[int, int, int, int]:
    """
    Return a tuple with 4 values denoting the floor and ceil bounds of the
    epicenter of the earthquake: floor_xE, floor_yE, ceil_xE, ceil_yE
    
    N: the number of persons
    persons: a list of N tuples (Xi, Yi, Si, Ri) denoting a person
    """
    # YOUR CODE HERE
    return 0, 0, 1, 1

def main():
    T = int(input())
    for _ in range(T):
        N = int(input())
        persons = []
        for _ in range(N):
            Xi, Yi, Si, Ri = input().split()
            persons.append((float(Xi), float(Yi), float(Si), Ri))
        floor_x_E, floor_y_E, ceil_x_E, ceil_y_E = solve(N, persons)
        print(floor_x_E, floor_y_E, ceil_x_E, ceil_y_E)

if __name__ == '__main__':
    main()
